lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1077,22 +1087,39 @@ public void unpublishObjectsExternally(Set<GUID> guids, int accessGroupId) throw
         for (Map<String, Object> hit : hitList) {
             @SuppressWarnings("unchecked")
             Map<String, Object> obj = (Map<String, Object>) hit.get("_source");
-            ObjectData item = buildObjectData(obj, pp.objectInfo, pp.objectKeys, 
-                    pp.objectData, pp.objectDataIncludes);
+            Map<String, ArrayList> highlightRes = (Map<String, ArrayList>) hit.get("highlight");
+            ObjectData item = buildObjectData(obj, editHighlightedKeys(highlightRes), pp);
             ret.add(item);
         }
         return ret;
     }
 
+    private Map<String, ArrayList> editHighlightedKeys(Map<String, ArrayList> highlightRes){
+        if(highlightRes == null){
+            return null;
+        }
+        Map<String, ArrayList> highlightResEdited = new HashMap<>();
+        for(String key : highlightRes.keySet()){
+            String newKey = key;
+            if(key.startsWith("key.")){
+                //truncate key. from key
+                newKey = key.substring(4);
+            }
+            highlightResEdited.put(newKey, highlightRes.get(key));
```

Comment:
What are the typical contents of a highlight map?

Reply:
highlight: [{"field" : "<em>matched term</em>"},{"field2:" : "<em>matched term</em>"}] for each object.

Reply:
is matched term a list? that's what it looks like in ObjectData

Reply:
the whole list is the list. The object may have multiple matches and not all in the same field

Reply:
If highlight in your comment above is the map key, why are you removing `key.` from the front? You really need to go through the list for each highlight key, go through the each map in the list, and edit those, since those are where the search keys are

Reply:
Sorry I just realized you were asking about the field stored in ObjectData. 
That goes: 
{"field1":["match1], "field2":["match1", "match2"]}
so the keys are the fields and matches show up in arrays.

Reply:
But it's the same structure from elastic, right? Currently the code just adds it to the ObjectData after stripping of `key.`:

```
Map<String, ArrayList> highlightRes = (Map<String, ArrayList>) hit.get("highlight");
ObjectData item = buildObjectData(obj, editHighlightedKeys(highlightRes), pp);
```

Reply:
Yes its the same as elastic. First pattern was wrong